README.md

```diff
@@ -169,56 +179,91 @@ You may customize the agent's behaviour by setting these environment variables.
+/!\ The address in the variables `MECH_TO_CONFIG` and `MECH_TO_SUBSCRIPTION` should be identical and correspond 
+to the address of the Mech contract.
 
-Now, you have two options to run the worker: as a standalone agent or as a service.
+If you want to run a legacy Mech, the `MECH_MARKETPLACE_ADDRESS` is optional. Otherwise this variable needs to be defined. 
+Furthermore, in the variable `MECH_TO_CONFIG`, the value corresponding to the key `is_marketplace_mech` should be set to true.
 
-### Option 1: Run the Mech as a standalone agent
+The rest of the common environment variables are present in the [service.yaml](https://github.com/valory-xyz/mech/blob/main/packages/valory/services/mech/service.yaml), which are customizable too.
 
-1. Ensure you have a file with a private key (`ethereum_private_key.txt`). You can generate a new private key file using the Open Autonomy CLI:
+#### Running the service 
+
+1. Ensure you have a file with the agent address and private key (`keys.json`). You can generate a new private key file using the Open Autonomy CLI:
+
+    ```bash
+    autonomy generate-key ethereum -n 1
+    ```
+
+2. Ensure that the variable `ALL_PARTICIPANTS` in the file `.1env` contains the same agent address as in `keys.json`:
 
    ```bash
-   autonomy generate-key ethereum 
+   ALL_PARTICIPANTS='["your_agent_address"]'
    ```
 
-2. From one terminal, run the agent:
+3. Run the service:
 
     ```bash
-    bash run_agent.sh
+    bash run_service.sh
     ```
 
-3. From another terminal, run the Tendermint node:
+4. Build the docker container: copy the four letters word in the folder `abci_build_{word}` in the `mech` folder and run the following:
 
-    ```bash
-    bash run_tm.sh
-    ```
+```
+cd mech
+autonomy deploy run --build-dir abci_build_{word} 
+```
 
-### Option 2: Run the Mech as an agent service
+where word is replaced by this four letters word.
 
-1. Ensure you have a file with the agent address and private key (`keys.json`). You can generate a new private key file using the Open Autonomy CLI:
+5. In a separate terminal, run the following to see the logs:
 
-    ```bash
-    autonomy generate-key ethereum -n 1
-    ```
+```
+docker logs -f mech{word}_abci_0
+```
+
+### Option 2: Run the Mech as a standalone agent
+
+#### Configuration of the agent
+
+First, you need to configure the agent, by adding parameters to the `packages/valory/agents/mech/aea-config.yaml` file. 
+The variables to change are the same as for the [service](#configuration-of-the-service). In this file you can also find other variables which can be customized.
+
+#### Running the agent 
 
-2. Ensure that the variable `ALL_PARTICIPANTS` in the file `.1env` contains the agent address from `keys.json`:
+
+1. Ensure you have a file with a private key (`ethereum_private_key.txt`). You can generate a new private key file using the Open Autonomy CLI:
 
    ```bash
-   ALL_PARTICIPANTS='["your_agent_address"]'
+   autonomy generate-key ethereum 
    ```
```

Comment:
This should happen before setting the env vars, otherwise, the user has no address to set.